packages/flutter/lib/src/widgets/form.dart

```diff
@@ -237,8 +237,17 @@ class FormState extends State<Form> {
 
   bool _validate() {
     bool hasError = false;
+    bool validateFormField(FormFieldState<dynamic> field) =>
+        !field.validate() || hasError;
+
     for (final FormFieldState<dynamic> field in _fields) {
-      hasError = !field.validate() || hasError;
+      if (widget.autovalidateMode == AutovalidateMode.onUserInteraction) {
+        if (_hasInteractedByUser && field._hasInteractedByUser.value) {
```

Comment:
also, should we remove _hasInteractedByUser here? It seems like _hasInteractedByUser will always be true if field._hasInteractedByUser.value is true..